dev-docs/HOW_THINGS_WORK.md

```diff
@@ -2,3 +2,341 @@
+triggers then copy the incorrect key onto the correct key in the keys table.
+This bug is fixed with `keys table is the only source if truth for public_key`
+Current Status:
+- Currently Phase1 is in rollout.
+- knife uses v0
+
+v1:
+- The user is not allowed to edit the public\_key via the /users endpoint,
+it must be managed via the /keys endpoint instead
+- chef-server-ctl uses v1
+
+## Keys table is the only source of truth for public\_key
+
+Phase1(Currently implemented):
+- Add a function for add\_user and update\_user that inserts only the sentinel
+value into the public\_key field in the users table.
+- The add and update triggers in keys\_update\_trigger.sql are present to maintain
+compatibility with older versions of chef-server.
+
+Phase2(To be implemented):
+Todo:
+- Delete the and and update triggers in keys\_update\_trigger.sql.
+- Delete the public\_key column in users (and perhaps clients) table.
+Notes:
+- This should be easier once all the users are on Phase1 release of chef-server.
+- The code will then only interact with the add\_user and update\_user functions in
+the back.
+- The upgrade path after Phase2 rollout will include upgrading to a release with
+Phase1 first.
+
+## Buildkite
+
+Buildkite is a platform for running fast, secure, and scalable continuous integration pipelines on your own infrastructure.
+Pipelines contain unit, integration, and other tests to help assess and validate your build.
+The first step to investigating errors is to check the build logs.
+
+The main pipelines for this repository are:
+* `[chef/chef-server:main] verify`
+* `[chef/chef-server:main] omnibus/adhoc`
+* `[chef/umbrella:main] chef-server`
+
+### [chef/chef-server:main] verify
+Verify pipeline runs all the unit tests. 
+A verify build is automatically triggered when changes to the branch are pushed and there is a pull request linked to it.
+The results of an automatically-triggerd verify build are linked to the pull request. If the build fails the pull request will be blocked.
+This build can also be triggered manually.
+
+### [chef/chef-server:main] omnibus/adhoc
+omnibus/adhoc pipeline runs the integration tests on different builds. Integration test covers the API endpoints of the project.
+The pipeline creates different build for different supported OS/environment and pushes the builds to jfrog artifactory.
+The integration tests scripts are tested against each of the different builds created.
+This is pipeline is automatically triggered every night to make sure that the main is always ready to ship.
+
+### [chef/umbrella:main] chef-server
+This pipeline is for end to end testing and creates different builds integrating with other projects of chef.
+These builds are packaged to replicate the different environment in which chef-server will be used by the customers.
+This pipeline is run on a nightly basis using the latest build from the current omnibus pipeline.
+
+## Using S3 instead of Bookshelf
```

Comment:
The HOW_THINGS_WORK.md has been split up into multipel docs. This section should be split into a new doc, bookshelf.md; this can be a subsection in that doc, maybe with placeholders for What It Does/How It Works?   And linked in dev-docs/README.md

Reply:
Done.